### ISSUE TITLE
PowerShell bootstrap script not working with PS 6

### ISSUE THREAD

**Issue by gep13** [cake-build/resources#21]
@kentcb commented on [Tue Dec 06 2016](https://github.com/cake-build/cake/issues/1393)

### What You Are Seeing?

When running the [bootstrap script](https://github.com/cake-build/example/blob/master/build.ps1) on PowerShell 6 (.NET core):
```
Method invocation failed because [System.Reflection.Assembly] does not contain a method named 'LoadWithPartialName'.
At C:\XXX\build.ps1:58 char:1
+ [Reflection.Assembly]::LoadWithPartialName("System.Security") | Out-N ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : MethodNotFound
```

`$PSVersionTable.PSVersion` gives me:

```
Major Minor Patch Label
----- ----- ----- -----
    6     0     0 alpha
```

### What is Expected?

Script should work with new versions of PS.

### What version of Cake are you using?

Haven't gotten that far yet (bootstrapping).

### Are you running on a 32 or 64 bit system?

64 bit.

### What environment are you running on?  Windows? Linux? Mac?

Windows 10, but the same behavior _should_ be present on other operating systems.

### Are you running on a CI Server?  If so, which one?

No

### How Did You Get This To Happen? (Steps to Reproduce)

1. Grab the bootstrap script
2. Attempt to execute it in a PS 6 session

### Output Log

N/A

**Comment by gep13** [comment]
@devlead as the resident PowerShell guru, do you have any thoughts on this one?

**Comment by netxph** [comment]
Just comment out
`[Reflection.Assembly]::LoadWithPartialName("System.Security") | Out-Null`

System.Security is already inside System.Core. I updated, to PowerShell 6 b3, and should be working now.

**Comment by gjonespf** [comment]
Made a quick patch for this for PS6, uses Get-FileHash in its place:
https://github.com/cake-build/resources/compare/master...gjonespf:xplat-md5
Worth a PR?

**Comment by devlead** [comment]
@gjonespf sure

**Comment by fleed** [comment]
Powershell Core 6.0 is out: [https://blogs.msdn.microsoft.com/powershell/2018/01/10/powershell-core-6-0-generally-available-ga-and-supported/](https://blogs.msdn.microsoft.com/powershell/2018/01/10/powershell-core-6-0-generally-available-ga-and-supported/).
Any plan to have first-class support for it?

**Comment by devlead** [comment]
Yes, just released 0.25.0 will give us the opportunity make bootstrappers cleaner, shouldn't be too hard to support it.

**Comment by devlead** [comment]
Hopefully the discussion and work going on in issue #53 will sort this too, that's the goal at least.
